x-pack/solutions/security/plugins/security_solution/server/lib/entity_analytics/privilege_monitoring/privilege_monitoring_data_client.ts

```diff
@@ -285,4 +310,70 @@ export class PrivilegeMonitoringDataClient {
 
     return this.opts.auditLogger?.log(event);
   }
+
+  // --- Privileged User Sync Orchestration ---
+  // These methods coordinate syncing users from monitoring sources.
+  public async syncAllIndexUsers() {
+    // get all monitoring index sources of type 'index
+    const indexSources: MonitoringEntitySourceDescriptor[] =
+      await this.monitoringIndexSourceClient.findByIndex();
+    this.log(
+      'debug',
+      `Found index sources for privilege monitoring:\n${JSON.stringify(indexSources, null, 2)}`
+    );
+    // get usernames from each index source
+    const allUserNames = new Set<string>();
+    for (const source of indexSources) {
+      const index = source.indexPattern ?? '';
+      const kuery =
+        typeof source.filter?.kuery === 'string' ? (source.filter.kuery as string) : undefined;
+      try {
+        // store / sync usernames in internal privileged users index
+        const usernames = await this.createUsersFromIndexSource(index, kuery);
+        usernames.forEach((username) => allUserNames.add(username));
+      } catch (error) {
+        this.log('error', `Failed to sync users from index ${index}: ${error}`);
+      }
+    }
+    await this.removeStaleIndexUsers(allUserNames);
+  }
+
+  public async createUsersFromIndexSource(indexName: string, kuery?: string): Promise<Set<string>> {
+    const query = kuery ? toElasticsearchQuery(fromKueryExpression(kuery)) : { match_all: {} };
+
+    const response = await this.esClient.search<{ user?: { name?: string } }>({
+      index: indexName,
+      _source: ['user.name'],
+      query,
+    });
```

Comment:
I think this deserves it's own function on the client.

We need to set a limit on this query as the default is 10, I think we will need to paginate happy to talk this though